### PR TITLE
[feat/fix/docs]: Fix bubble_sort2.c implementation bug and some documentation/structure improvements

### PR DESCRIPTION
**Pull request by northernSage**
#### Description of Change

As of now, the implementation of the bubble-sort algorithm in [bubble_sort2.c](https://github.com/TheAlgorithms/C/blob/master/sorting/bubble_sort_2.c) is not working.
```c
int i, arraySort[MAX] = {0}, isSort = FALSE, changePlace;
 // ...
while (isSort) { // sorting code... }
```
The above snippet illustrates the problem. The flag `isSort` is initially set to [`FALSE`](https://github.com/TheAlgorithms/C/blob/master/sorting/bubble_sort_2.c#L5) and, consequently, the loop where the array would be sorted never runs. Other than that, we can also see _camelCase_ being used and the lack of a separate test function and documentation as per `CONTRIBUTING.md` [instructions](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#typical-structure-of-a-program).

#### Checklist

- [x] replace camelCase by snake_case
- [x] add a separate test function
- [x] fix bug with `isSort` flag
- [x] document code
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: sorting/bubble_sort_2.c [hunk 1]
```diff
@@ -1,46 +1,84 @@
+/**
+ * @file
+ * @brief implementation of [Bubble sort](https://en.wikipedia.org/wiki/Bubble_sort) algorithm
+ * @details
+ * worst-case: O(n^2)
+ * best-case: O(n)
+ * average-complexity: O(n^2)
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:7]:
```suggestion
 * average-complexity: O(n^2)

 * @author Unknown author
 * @author [Gabriel Fioravante](https://github.com/northernSage)
```

File: sorting/bubble_sort_2.c [hunk 2]
```diff
@@ -1,46 +1,84 @@
...
+ */
+
 #include <stdio.h>
 #include <stdlib.h>
+#include <assert.h>
+#include <stdbool.h>
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:13]: Please add a brief description of what the library adds (see the example below).
```c++
#include <iostream>   /// for IO operations
#include <vector>    /// for std::vector
```

File: sorting/bubble_sort_2.c [hunk 3]
```diff
@@ -1,46 +1,84 @@
...
+
 #define MAX 20
-#define TRUE 1
-#define FALSE 0
 
-int main()
+/**
+ * Bubble sort implementation
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:18]:
```suggestion
 * @brief Bubble sort implementation
```

File: sorting/bubble_sort_2.c [hunk 4]
```diff
@@ -1,46 +1,84 @@
...
+ * @param array_sort the array to be sorted
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:19]:
```suggestion
 * @param array_sort the array to be sorted
 * @returns void
```

File: sorting/bubble_sort_2.c [hunk 5]
```diff
@@ -1,46 +1,84 @@
...
+}
 
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+	test(); // call test routine
     return EXIT_SUCCESS;
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:83]:
```suggestion
    test();  // run self-test implementations
    return EXIT_SUCCESS;
```

File: sorting/bubble_sort_2.c [hunk 6]
```diff
@@ -1,46 +1,84 @@
...
+                 * so we reset the flag to keep ordering
+                 * until no swap operations are executed */
+                is_sorted = false;
             }
         }
     }
+}
+
+/**
+ * @brief Test function
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:52]:
```suggestion
 * @brief Test implementations
```

File: sorting/bubble_sort_2.c [hunk 7]
```diff
@@ -1,46 +1,84 @@
...
 #define MAX 20
-#define TRUE 1
-#define FALSE 0
 
-int main()
+/**
+ * Bubble sort implementation
+ * @param array_sort the array to be sorted
+ */
+void bubblesort(int* array_sort)
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:21]:
```suggestion
void bubble_sort(int* array_sort)
```

File: sorting/bubble_sort_2.c [hunk 8]
```diff
@@ -1,46 +1,84 @@
...
+    /* populate our test array with 
+     * random integer numbers */
+    for (int i = 0; i < MAX; i++)
+    {
+        array_sort[i] = rand() % 101;
+    }
 
-    /* See if it works */
+    /* sort array */
+    bubblesort(array_sort);
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:67]:
```suggestion
    bubble_sort(array_sort);
```

File: sorting/bubble_sort_2.c (hunk 5 above)
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:83]:
```suggestion
    return 0;
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Well done.

File: sorting/bubble_sort_2.c (hunk 2 above)
Comment (northernSage) [reply sorting/bubble_sort_2.c]: Oops, actually forgot to remove `stdio.h`. Thanks for the tip 👍🏻

File: sorting/bubble_sort_2.c [hunk 9]
```diff
@@ -1,46 +1,87 @@
...
+ * @author Unknown author
+ * @author [Gabriel Fioravante](https://github.com/northernSage)
+ */
 
-    /* For example
-    Insertion random values in array to test
-    */
+#include <stdlib.h>   // for rand() calls
+#include <assert.h>   // for testing assert(<expr>)
+#include <stdbool.h>  // for boolean values: true, false
```
Comment (Panquesito7) [inline sorting/bubble_sort_2.c:15]:
```suggestion
#include <stdlib.h>   /// for rand() calls
#include <assert.h>   /// for assert(<expr>)
#include <stdbool.h>  /// for boolean values: true, false  
```

**Review by Panquesito7 — APPROVED** [review]
LGTM. Thank you for your contribution! 👍 🎉

**Review by ayaankhan98 — APPROVED** [review]

**Pull request merged**